packages/search/lib/commands/CURSOR_DEL.ts

```diff
@@ -3,8 +3,8 @@ import { SimpleStringReply, Command, RedisArgument } from '@redis/client/dist/li
 export default {
   FIRST_KEY_INDEX: undefined,
   IS_READ_ONLY: true,
-  transformArguments(index: RedisArgument, cursorId: RedisArgument) {
-    return ['FT.CURSOR', 'DEL', index, cursorId];
+  transformArguments(index: RedisArgument, cursorId: number) {
```

Comment:
cursorId can go outside the float64 range.. string has to be supported, number is nice to have

Reply:
ok, was a copy form v4 which used number.  trying to remember why I changed it from what you did originally, I suspect number might be used elsewhere and needs to be fixed, will look into that.

Reply:
so I'm also unconvinced that "number" is nice to have, its a "number", but in practice, cursors should be opaque, and we shouldn't encourage users to think of it as a number, so thinking that RedisArgument as is, should be enough (as you had originally)

Reply:
so the issue here is really due to AGGREGATE_WITHCURSOR (where cursor id is returned to user).  Changed it to be a NumberReply there.  I think before I ran into some problems with tests, will have to redo and see what the problem was.

Reply:
i.e. the problem is that AGGREGATE_WITHCURSOR when cursor is a NumberReply will default to returning as a number.

instead returning NumberReply (from WITHCURSOR) for cursor and taking an UnwrapReply<NumbeReply> for cursor in cursor read/del and its up to the user to therefore have a proper number conversion setup.

Honestly, not happy with it. I'd want to force the NumberReply to be treated as a string and not as a number in WITHCURSOR (and then can force a string in cursor functions)

Reply:
"I'd want to force the NumberReply to be treated as a string and not as a number in WITHCURSOR (and then can force a string in cursor functions)" - I'd want the server to reply with a string and not a number..
The server not gonna change, we should probably support both.

Reply:
so we have the NumberReply that can be both.  With that said, I'd think we should document this (i.e. we strongly reccomend one overrides the type mapping for NumberReplies when using the WITHCURSOR function).

A better solution (which I dont know how to implement) would be to be able to tag commands with a default type mapping that can only override default (blank) client type mapping, but not type mapping that the user specifies.

i.e. by default we might say a NumberReply maps to a a js number, but for some commands we can say it maps to string, and the client would then just do the right thing by default always without user intervention required.